doc: a handful of doc improvements

jasnell opened the pull request:
Four separate doc improvements covering various issues. I've bundled these up
as separate commits within a single PR to make it easier to land.

sarathms committed:
Documentation update about Buffer initialization

fixes #7230

chrisneave committed:
docs: Fix default options for fs.createWriteStream()

The documentation for createWriteStream() references an
'encoding' property that has a default value of null. However,
this property is never referenced by createWriteStream() or
WritableState(). Instead a 'defaultEncoding' property is
referenced in WritableState() with a default of 'utf8' if no value
is supplied.

This fix updates the documentation to rename the 'encoding'
property to 'defaultEncoding' and indicate its default value of
'utf8'.

jasnell committed:
doc: improve http.abort description

Per #4409, the documentation on http.abort is a bit lacking.
This provides a slight improvement.

jasnell committed:
doc: mention that mode is ignored if file exists

per: https://github.com/joyent/node/issues/6847

jasnell committed:
doc: add CleartextStream deprecation notice

Per: https://github.com/joyent/node/issues/8017

jasnell commented:
@joyent/node-tsc ... can I get a quick review on these doc changes?

trevnorris commented on doc/api/buffer.markdown:
```diff
@@ -64,7 +64,7 @@ It can be constructed in a variety of ways.
 Allocates a new buffer of `size` octets. Note, `size` must be no more than
 [kMaxLength](smalloc.html#smalloc_smalloc_kmaxlength). Otherwise, a `RangeError`
-will be thrown here.
+will be thrown here. Unlike `ArrayBuffers`, the underlying memory for buffers is not initialized. So the contents of a newly created `Buffer` is unknown. Use `buf.fill(0)`to initialize a buffer to zeroes.
```
Reads well. I'd suggest word wrapping the same as above.

trevnorris commented:
One nit. Otherwise LGTM.

jasnell committed:
doc: fix line wrapping in buffer.markdown

Fix the line wrapping in buffer.markdown

jasnell commented:
@trevnorris ... better? (I'll squash down the commits before landing)

trevnorris commented:
looks great.

jasnell committed:
doc: add a note about readable in flowing mode

Original: https://github.com/joyent/node/pull/8682

Slightly modified version of the original PR (#8682) to add
appropriate line wrapping and fix a couple of grammar nits.

jasnell commented:
Landed in b4a0ca2, 1b027ee, ba6c086, e7f4237, 70efdf3, 500ff56, and b436e59
